base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
Yes, that looks fine. But the problem is that if you want to implement `copy!` for *non*-arrays, you need to use `push!` AFAICT. Which is not correct for arrays, hence this comment thread.